src/charm.py

```diff
@@ -37,6 +38,30 @@
 logger = logging.getLogger(__name__)
 
 
+def evaluate_status(func: Callable) -> Callable:
```

Comment:
If I have to do something like this, I will probably go the route of a simple function that will be called in several places. e.g. `self.asses_status()`

Not saying that this is wrong or worst solution. I asked in ops channel about alternative for [atexit](https://github.com/juju/charm-helpers/blob/82267ac050e55b0348542858bc2c94ca69ef5895/charmhelpers/core/hookenv.py#L1337-L1341) function.

Reply:
My reasoning here is that you want to always call this at the end of the event handler. Decorator accomplishes two things:
* ensures that assessment is done after the handler is executed
* Makes it easier to spot that some additional "status assessment" is done

(I see that suggestion in `ops` channel also went in the direction of wrapper/decorator)

Reply:
Yes, but they suggested to have `try except`, what we are not doing. If `_on_config_changed` failed status will not be changed, what completely make sense.

In my option there could be event `atexit` which will be run after any successful event.

Reply:
I agree with you that `atexit` equivalent would be handy...but since we don't have one I think this is acceptable workaround.